orion/pkg/repos/semester_repo.go

```diff
@@ -169,35 +170,53 @@ func (sr *semesterRepo) Update(semesterId string, semester domains.Semester) err
 	return handleSqlExecResult(execResult, 1, "semester was not updated")
 }
 
-func (sr *semesterRepo) Delete(semesterId string) error {
-	statement := "DELETE FROM semesters WHERE semester_id=?"
-	stmt, err := sr.db.Prepare(statement)
+func (sr *semesterRepo) Publish(semesterIds []string) error {
+	var errorString string
+
+	tx, err := sr.db.Begin()
 	if err != nil {
 		return err
 	}
-	defer stmt.Close()
-
-	execResult, err := stmt.Exec(semesterId)
+	stmt, err := tx.Prepare("UPDATE semesters SET published_at=? WHERE semester_id=? AND published_at IS NULL")
 	if err != nil {
 		return err
 	}
-	return handleSqlExecResult(execResult, 1, "semester was not deleted")
+	defer stmt.Close()
+
+	now := time.Now().UTC()
+	for _, semesterId := range semesterIds {
+		execResult, err := stmt.Exec(now, semesterId)
+		if err != nil {
+			errorString = appendError(errorString, semesterId, err)
+			continue
+		}
+		err1 := handleSqlExecResult(execResult, 0, "") // more than 1 row affected
+		err2 := handleSqlExecResult(execResult, 1, "") // 0 or multiple rows affected
+		if err1 != nil && err2 != nil {
+			errorString = appendError(errorString, semesterId, errors.New("multiple rows changed"))
+		}
```

Comment:
as discussed, you don't need to use handlSqlExecResult to check if rows get updated. We are already appending error strings if they are not being correctly added.